File: src/main/python/net/coderextreme/data/rgb_alpha.py
print('<!--')
import x3d
print('-->')
X3D0 = x3d.X3D()
X3D0.profile = "Interchange"
X3D0.version = "4.1"
head1 = x3d.head()
meta2 = x3d.meta()
meta2.name = "title"
meta2.content = "rgb_alpha.x3d"

head1.children.append(meta2)
meta3 = x3d.meta()
meta3.name = "Image"
meta3.content = "rgb_alpha-front.jpg"

head1.children.append(meta3)
meta4 = x3d.meta()
meta4.name = "Image"
meta4.content = "rgb_alpha-rear.jpg"

head1.children.append(meta4)
meta5 = x3d.meta()
meta5.name = "Image"
meta5.content = "rgb_alpha-top.jpg"

head1.children.append(meta5)
meta6 = x3d.meta()
meta6.name = "Image"
meta6.content = "rgb_alpha-bottom.jpg"

head1.children.append(meta6)
meta7 = x3d.meta()
meta7.name = "reference"
meta7.content = "http://www.nist.gov/vrml.html"

head1.children.append(meta7)
meta8 = x3d.meta()
meta8.name = "reference"
meta8.content = "http://www.itl.nist.gov/div897/ctg/vrml/vrml.html"

head1.children.append(meta8)
meta9 = x3d.meta()
meta9.name = "creator"
meta9.content = "http://www.itl.nist.gov/div897/ctg/vrml/members.html"

head1.children.append(meta9)
meta10 = x3d.meta()
meta10.name = "disclaimer"
meta10.content = "This file was provided by the National Institute of Standards and Technology, and is part of the X3D Conformance Test Suite, available at http://www.nist.gov/vrml.html The information contained within this file is provided for use in establishing conformance to the ISO VRML97 Specification. Conformance to this test does not imply recommendation or endorsement by the National Institute of Standards and Technology. This software can be redistributed and/or modified freely provided that any derivative works bear some notice that they are derived from it, and any modified versions bear some notice that they have been modified."

head1.children.append(meta10)
meta11 = x3d.meta()
meta11.name = "info"
meta11.content = "Correct definition and compliance of this conformance scene is maintained by the X3D Working Group, https://www.web3d.org/working-groups/x3d"

head1.children.append(meta11)
meta12 = x3d.meta()
meta12.name = "translator"
meta12.content = "Michael Kass NIST, Don Brutzman NPS"

head1.children.append(meta12)
meta13 = x3d.meta()
meta13.name = "translated"
meta13.content = "21 January 2001"

head1.children.append(meta13)
meta14 = x3d.meta()
meta14.name = "modified"
meta14.content = "Mon, 09 Feb 2026 07:12:59 GMT"

head1.children.append(meta14)
meta15 = x3d.meta()
meta15.name = "description"
meta15.content = "Test browser ability to map a RGB plus alpha opacity to geometry. A checkerboard of four colored squares: lower left (red), lower right (transparent), uppser left (transparent) and upper right (red) map onto the faces of all geometry. For the sphere, the texture should cover the entire surface, and wrap counterclockwise from the back of the sphere. For the cone, the texture should wrap counterclockwise (from above) starting at the back of the cone. A circle cutout of the texture is applied right side up to the base of the cone when the cone is tilted toward the -z axis. For the cylinder, the texture should wrap counterclockwise (from above) starting at the back of the cylinder. A circle cutout of the texture is applied right side up to the top and bottom caps of the cylinder. For the box, the texture should be applied right side up in its entirety to each face of the box."

head1.children.append(meta15)
meta16 = x3d.meta()
meta16.name = "identifier"
meta16.content = "https://www.web3d.org/x3d/content/examples/ConformanceNist/Appearance/PixelTexture/rgb_alpha.x3d"

head1.children.append(meta16)
meta17 = x3d.meta()
meta17.name = "generator"
meta17.content = "x3d-tidy V3.0.2, https://www.npmjs.com/package/x3d-tidy"

head1.children.append(meta17)

X3D0.head = head1
Scene18 = x3d.Scene()
NavigationInfo19 = x3d.NavigationInfo()
NavigationInfo19.type = ["EXAMINE","WALK","FLY","ANY"]

Scene18.children.append(NavigationInfo19)
Group20 = x3d.Group()
Transform21 = x3d.Transform()
Transform21.translation = [6.14221,0.0694613,-0.000999451]
Shape22 = x3d.Shape()
Appearance23 = x3d.Appearance()
Material24 = x3d.Material()

Appearance23.material = Material24
PixelTexture25 = x3d.PixelTexture(DEF="RgbOpacityCheckerboard")
PixelTexture25.image = [2,2,4,-16776961,-65536,-65536,-16776961]

Appearance23.texture = PixelTexture25

Shape22.appearance = Appearance23
Box26 = x3d.Box()

Shape22.geometry = Box26

Transform21.children.append(Shape22)

Group20.children.append(Transform21)
Transform27 = x3d.Transform()
Transform27.translation = [-4.85443,0.0694381,-0.00149918]
Shape28 = x3d.Shape()
Appearance29 = x3d.Appearance()
Material30 = x3d.Material()

Appearance29.material = Material30
PixelTexture31 = x3d.PixelTexture(USE="RgbOpacityCheckerboard")

Appearance29.texture = PixelTexture31

Shape28.appearance = Appearance29
Sphere32 = x3d.Sphere()

Shape28.geometry = Sphere32

Transform27.children.append(Shape28)

Group20.children.append(Transform27)
Transform33 = x3d.Transform()
Transform33.translation = [-1.47341,0.036672,-0.00175095]
Shape34 = x3d.Shape()
Appearance35 = x3d.Appearance()
Material36 = x3d.Material()

Appearance35.material = Material36
PixelTexture37 = x3d.PixelTexture(USE="RgbOpacityCheckerboard")

Appearance35.texture = PixelTexture37

Shape34.appearance = Appearance35
Cone38 = x3d.Cone()

Shape34.geometry = Cone38

Transform33.children.append(Shape34)

Group20.children.append(Transform33)
Transform39 = x3d.Transform()
Transform39.translation = [2.31094,0.0694206,-0.00187683]
Shape40 = x3d.Shape()
Appearance41 = x3d.Appearance()
Material42 = x3d.Material()

Appearance41.material = Material42
PixelTexture43 = x3d.PixelTexture(USE="RgbOpacityCheckerboard")

Appearance41.texture = PixelTexture43

Shape40.appearance = Appearance41
Cylinder44 = x3d.Cylinder()

Shape40.geometry = Cylinder44

Transform39.children.append(Shape40)

Group20.children.append(Transform39)

Scene18.children.append(Group20)

X3D0.Scene = Scene18
f = open("../data/rgb_alpha.new.python.x3d", mode="w", encoding="utf-8")
f.write(X3D0.XML())
f.close()
f = open("../data/rgb_alpha.new.python.x3dv", mode="w", encoding="utf-8")
f.write(X3D0.VRML())
f.close()
f = open("../data/rgb_alpha.new.python.x3dj", mode="w", encoding="utf-8")
f.write(X3D0.JSON())
f.close()

File: src/main/cplusplus/net/coderextreme/data/rgb_alpha.cpp
#ifdef WIN32
#define FALSE false
#define TRUE true
#define WINAPI
#define AFX_EXT_CLASS
#define EXPORT32
#endif
#define False false
#define True true
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/SphereExample/SphereExample/pch.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/SphereExample/SphereExample/framework.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/SphereExample/include/XML_PARSER.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/SphereExample/include/X3DLib.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/SphereExample/include/glMath.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/SphereExample/include/glut.h"
int main(int argc, char ** argv) {
X3D* X3D0 = new X3D();
X3D0->setProfile(CString("Interchange"));
X3D0->setVersion(CString("3.0"));
head* head1 = new head();
meta* meta2 = new meta();
meta2->setName(CString("title"));
meta2->setContent(CString("rgb_alpha.x3d"));
head1->addMeta(meta2);

meta* meta3 = new meta();
meta3->setName(CString("Image"));
meta3->setContent(CString("rgb_alpha-front.jpg"));
head1->addMeta(meta3);

meta* meta4 = new meta();
meta4->setName(CString("Image"));
meta4->setContent(CString("rgb_alpha-rear.jpg"));
head1->addMeta(meta4);

meta* meta5 = new meta();
meta5->setName(CString("Image"));
meta5->setContent(CString("rgb_alpha-top.jpg"));
head1->addMeta(meta5);

meta* meta6 = new meta();
meta6->setName(CString("Image"));
meta6->setContent(CString("rgb_alpha-bottom.jpg"));
head1->addMeta(meta6);

meta* meta7 = new meta();
meta7->setName(CString("reference"));
meta7->setContent(CString("http://www.nist.gov/vrml.html"));
head1->addMeta(meta7);

meta* meta8 = new meta();
meta8->setName(CString("reference"));
meta8->setContent(CString("http://www.itl.nist.gov/div897/ctg/vrml/vrml.html"));
head1->addMeta(meta8);

meta* meta9 = new meta();
meta9->setName(CString("creator"));
meta9->setContent(CString("http://www.itl.nist.gov/div897/ctg/vrml/members.html"));
head1->addMeta(meta9);

meta* meta10 = new meta();
meta10->setName(CString("disclaimer"));
meta10->setContent(CString("This file was provided by the National Institute of Standards and Technology, and is part of the X3D Conformance Test Suite, available at http://www.nist.gov/vrml.html The information contained within this file is provided for use in establishing conformance to the ISO VRML97 Specification. Conformance to this test does not imply recommendation or endorsement by the National Institute of Standards and Technology. This software can be redistributed and/or modified freely provided that any derivative works bear some notice that they are derived from it, and any modified versions bear some notice that they have been modified."));
head1->addMeta(meta10);

meta* meta11 = new meta();
meta11->setName(CString("info"));
meta11->setContent(CString("Correct definition and compliance of this conformance scene is maintained by the X3D Working Group, https://www.web3d.org/working-groups/x3d"));
head1->addMeta(meta11);

meta* meta12 = new meta();
meta12->setName(CString("translator"));
meta12->setContent(CString("Michael Kass NIST, Don Brutzman NPS"));
head1->addMeta(meta12);

meta* meta13 = new meta();
meta13->setName(CString("translated"));
meta13->setContent(CString("21 January 2001"));
head1->addMeta(meta13);

meta* meta14 = new meta();
meta14->setName(CString("modified"));
meta14->setContent(CString("16 January 2011"));
head1->addMeta(meta14);

meta* meta15 = new meta();
meta15->setName(CString("description"));
meta15->setContent(CString("Test browser ability to map a RGB plus alpha opacity to geometry. A checkerboard of four colored squares: lower left (red), lower right (transparent), uppser left (transparent) and upper right (red) map onto the faces of all geometry. For the sphere, the texture should cover the entire surface, and wrap counterclockwise from the back of the sphere. For the cone, the texture should wrap counterclockwise (from above) starting at the back of the cone. A circle cutout of the texture is applied right side up to the base of the cone when the cone is tilted toward the -z axis. For the cylinder, the texture should wrap counterclockwise (from above) starting at the back of the cylinder. A circle cutout of the texture is applied right side up to the top and bottom caps of the cylinder. For the box, the texture should be applied right side up in its entirety to each face of the box."));
head1->addMeta(meta15);

meta* meta16 = new meta();
meta16->setName(CString("identifier"));
meta16->setContent(CString("https://www.web3d.org/x3d/content/examples/ConformanceNist/Appearance/PixelTexture/rgb_alpha.x3d"));
head1->addMeta(meta16);

meta* meta17 = new meta();
meta17->setName(CString("generator"));
meta17->setContent(CString("Vrml97ToX3dNist, http://ovrt.nist.gov/v2_x3d.html"));
head1->addMeta(meta17);

meta* meta18 = new meta();
meta18->setName(CString("generator"));
meta18->setContent(CString("X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit"));
head1->addMeta(meta18);

meta* meta19 = new meta();
meta19->setName(CString("license"));
meta19->setContent(CString("../../license.html"));
head1->addMeta(meta19);

X3D0->setHead(head1);

Scene* Scene20 = new Scene();
NavigationInfo* NavigationInfo21 = new NavigationInfo();
NavigationInfo21->setType(new CString[4]{CString("EXAMINE"), CString("WALK"), CString("FLY"), CString("ANY")}, 4);
Scene20->addChild(NavigationInfo21);

Group* Group22 = new Group();
Transform* Transform23 = new Transform();
Transform23->setTranslation(new float[3]{6.14221,0.0694613,-0.000999451});
Shape* Shape24 = new Shape();
Appearance* Appearance25 = new Appearance();
Material* Material26 = new Material();
Appearance25->addChild(Material26);

PixelTexture* PixelTexture27 = new PixelTexture();
PixelTexture27->setDEF(CString("RgbOpacityCheckerboard"));
PixelTexture27->setImage(new int32_t[7]{2,2,4,-16776961,-65536,-65536,-16776961});
Appearance25->addChild(PixelTexture27);

Shape24->addChild(Appearance25);

Box* Box28 = new Box();
Shape24->setGeometry(Box28);

Transform23->addChild(Shape24);

Group22->addChild(Transform23);

Transform* Transform29 = new Transform();
Transform29->setTranslation(new float[3]{-4.85443,0.0694381,-0.00149918});
Shape* Shape30 = new Shape();
Appearance* Appearance31 = new Appearance();
Material* Material32 = new Material();
Appearance31->addChild(Material32);

PixelTexture* PixelTexture33 = new PixelTexture();
PixelTexture33->setUSE(CString("RgbOpacityCheckerboard"));
Appearance31->addChild(PixelTexture33);

Shape30->addChild(Appearance31);

Sphere* Sphere34 = new Sphere();
Shape30->setGeometry(Sphere34);

Transform29->addChild(Shape30);

Group22->addChild(Transform29);

Transform* Transform35 = new Transform();
Transform35->setTranslation(new float[3]{-1.47341,0.036672,-0.00175095});
Shape* Shape36 = new Shape();
Appearance* Appearance37 = new Appearance();
Material* Material38 = new Material();
Appearance37->addChild(Material38);

PixelTexture* PixelTexture39 = new PixelTexture();
PixelTexture39->setUSE(CString("RgbOpacityCheckerboard"));
Appearance37->addChild(PixelTexture39);

Shape36->addChild(Appearance37);

Cone* Cone40 = new Cone();
Shape36->setGeometry(Cone40);

Transform35->addChild(Shape36);

Group22->addChild(Transform35);

Transform* Transform41 = new Transform();
Transform41->setTranslation(new float[3]{2.31094,0.0694206,-0.00187683});
Shape* Shape42 = new Shape();
Appearance* Appearance43 = new Appearance();
Material* Material44 = new Material();
Appearance43->addChild(Material44);

PixelTexture* PixelTexture45 = new PixelTexture();
PixelTexture45->setUSE(CString("RgbOpacityCheckerboard"));
Appearance43->addChild(PixelTexture45);

Shape42->addChild(Appearance43);

Cylinder* Cylinder46 = new Cylinder();
Shape42->setGeometry(Cylinder46);

Transform41->addChild(Shape42);

Group22->addChild(Transform41);

Scene20->addChild(Group22);

X3D0->setScene(Scene20);

}
